frontend/packages/integration-tests-cypress/README.md

```diff
@@ -0,0 +1,50 @@
+#### Getting Started
+- [What is Cypress?](https://www.youtube.com/watch?v=dr10Z-HpsCQ) (video)
+- [Cypress in a Nutshell](https://www.youtube.com/watch?v=LcGHiFnBh3Y) (video)
+
+#### Best Practices
+- We are switching over our test ids from `< ... data-test-id=".."/>` to using the Cypress preferred 
+`< ... data-test=".."/>`.  This allows us to better take advantage of certain Cypress tooling, like the 
+[Selector Playground](https://docs.cypress.io/guides/core-concepts/test-runner.html#Selector-Playground) 
+- Use [Cypress's Best Practices for Selecting Elements](https://docs.cypress.io/guides/references/best-practices.html)
+
+#### Migrating Protractor tests to Cypress
+
+When migrating a test suite from Protractor to Cypress, the following steps are recommended:
+1. Create the new test suite in Cypress
+2. If you need to create a new test id use the `data-test` attribute and the `cy.getByTestID()` helper method.
+If you need to access the legacy `data-test-id` attribute, use the `cy.getByLegacyTestID()` helper method.
+3. Remove test suite from Protractor
+
+#### Directory Structure
+```
+frontend/packages/integration-tests-cypress/
+├── cy-command-additions    <--- add commands to Cypress 'cy.' global
+│   ├── index.ts
+│   ├── nav.ts
+│   ├── project.ts
+│   ├── README.md
+│   └── selectors.ts
+├── fixtures                <--- mock data
+│   └── example.json
+├── tests                   <--- test suites
+│   ├── crud
+│   │   └── namespace-crud.cyspec.ts
```

Comment:
Do we need `cyspec` because it clashes with the unit tests? It would be nice to just use `spec` here and fix the pattern.

Reply:
yes, it was conflicting with unit tests

Reply:
> yes, it was conflicting with unit tests

Can we fix the pattern for unit tests instead?

Reply:
Hi @spadgett, I added `..integration-tests-cypress` to the `transformIgnorePatterns`:
```
 "transformIgnorePatterns": [
      "<rootDir>/node_modules/(?!(lodash-es|@console|@novnc|@spice-project)/.*)",
      "<rootDir>/packages/integration-tests-cypress/*"
    ],
    "testRegex": ".*\\.spec\\.(ts|tsx|js|jsx)$",
```
but still get the following error, so it seems like jest is not ignoring the files:
```
Summary of all failing tests
 FAIL  packages/integration-tests-cypress/tests/monitoring/monitoring.spec.ts
  ● Test suite failed to run
   /home/dtaylor/repos/openshift/origin/src/github.com/openshift/console/frontend/packages/integration-tests-cypress/tests/monitoring/monitoring.spec.ts:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import { testName } from '../../support';
```
Based on `testRegex`, we could do something like `namespace-crud.spec.cy.ts` or something similar.

Reply:
Looks like a bad regex. Maybe

```
<rootDir>/packages/integration-tests-cypress/.*
```

Reply:
I tried that initially (and just retried), no difference.  I think it has something to do with [this](https://github.com/facebook/jest/issues/2081)

Reply:
@dtaylor113 try this:

```json
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/packages/integration-tests-cypress"
    ],
```